### PR TITLE
types: add context flags with container flag to event

### PR DESCRIPTION
**Pull request by AlonZivony**
## Initial Checklist

- [ ] There is an issue describing the need for this PR.
- [x] Git log contains summary of the change.
- [x] Git log contains motivation and context of the change.
- [ ] If part of an EPIC, PR git log contains EPIC number.
- [ ] If part of an EPIC, PR was added to EPIC description.

## Description (git log)

Flags will enable to pass boolean information about the context from
the kernel.
Add the container context from the kernel in the new flags.

## Type of change

- [x] Bug fix (non-breaking change fixing an issue, preferable).
- [ ] Quick fix (minor non-breaking change requiring no issue, use with care)
- [ ] Code refactor (code improvement and/or code removal)
- [ ] New feature (non-breaking change adding functionality).
- [ ] Breaking change (cause existing functionality not to work as expected).

## Final Checklist:

Pick "Bug Fix" or "Feature", delete the other and mark appropriate checks.

- [x] I have made corresponding changes to the documentation.
- [x] My code follows the style guidelines (C and Go) of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented all functions/methods created explaining what they do.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix, or feature, is effective.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published before.

## Git Log Checklist:

My commits logs have:

- [x] Subject starts with "subsystem|file: description".
- [x] Do not end the subject line with a period.
- [x] Limit the subject line to 50 characters.
- [x] Separate subject from body with a blank line.
- [x] Use the imperative mood in the subject line.
- [x] Wrap the body at 72 characters.
- [x] Use the body to explain what and why instead of how.

### CODE REVIEW

File: types/trace/trace.go [hunk 1]
```diff
@@ -12,50 +12,59 @@ import (
...
+	if e.ContextFlags.ContainerStarted {
 		return ContainerOrigin
-	} else {
-		return HostOrigin
 	}
+	if e.ContainerID != "" {
+		return ContainerInitOrigin
+	}
+	return HostOrigin
 }
```
Comment (yanivagman) [inline types/trace/trace.go:68]: This function will break until we merge the container flags PR.
We should consider moving it out of types package.

Comment (AlonZivony) [reply types/trace/trace.go:68]: So me and @rafaeldtinoco saw that the bug wasn't from the container flag PR, but from unreliable tests.
The problem was fixed in the cb65c6ac commit.
This means we can safely merge now the 2 PRs for the container flag, so removing the `Origin` change isn't a problem.

Comment (yanivagman) [reply types/trace/trace.go:68]: Cool! What is this commit (cb65c6ac) exactly? I can't find it looking at main

Comment (NDStrahilevitz) [reply types/trace/trace.go:68]: > Cool! What is this commit (cb65c6ac) exactly? I can't find it looking at main

It's [this](https://github.com/aquasecurity/tracee/commit/cb56c6ace4cdb9b8a3f4b6aa74426deece708662).

Just to note though I think this change would further break old recordings on new `tracee-rules` versions.

**Review by yanivagman — APPROVED** [review]
This PR is actually the same as #2007 which was later reverted in #2027, but now discovered not to be causing the issue we thought it was (https://github.com/aquasecurity/tracee/pull/2041/files#r935748225).
So shouldn't be a problem to merge it again now.

**Pull request merged**